ratelimit_test.go

```diff
@@ -125,6 +126,32 @@ var takeTests = []struct {
 	}},
 }}
 
+var availTests = []struct {
+	capacity     int64
+	fillInterval time.Duration
+	take         int64
+	sleep        time.Duration
+
+	expectCountAfterTake  int64
+	expectCountAfterSleep int64
+}{
+	{
+		capacity:              5,
+		fillInterval:          time.Second,
+		take:                  5,
+		sleep:                 time.Second,
+		expectCountAfterTake:  0,
+		expectCountAfterSleep: 1,
+	}, {
+		capacity:              2,
+		fillInterval:          time.Second,
+		take:                  1,
+		sleep:                 time.Second,
```

Comment:
I'd like to see at least one test that tests that available returns the correct thing when not called on a fillInterval boundary.